pkg/util/xorm/dialect_spanner.go

```diff
@@ -0,0 +1,284 @@
+//go:build enterprise || pro
+
+package xorm
+
+import (
+	"strings"
+
+	_ "github.com/googleapis/go-sql-spanner"
+	"xorm.io/core"
+)
+
+func init() {
+	core.RegisterDriver("spanner", &spannerDriver{})
+	core.RegisterDialect("spanner", func() core.Dialect { return &spanner{} })
+}
+
+// https://cloud.google.com/spanner/docs/reference/standard-sql/lexical#reserved_keywords
+var spannerReservedKeywords = map[string]bool{
```

Comment:
Very nitpicky (so feel free to ignore) but normally if you need to emulate a set you can just use empty struct as a map value:
```suggestion
var spannerReservedKeywords = map[string]struct{}{
```

Reply:
struct{} also takes [no space](https://dave.cheney.net/2014/03/25/the-empty-struct) at all compared to bool.

Reply:
On the other hand, using bools leads to simpler usage (like `if spannerReservedKeywords[word] { .. }`) and at this size of map, memory saving isn't very big.